scriptworker/test/test_task.py

```diff
@@ -297,14 +298,51 @@ def test_prepare_to_run_task(context):
         return fake_proc
 
     mocker.patch.object(asyncio, 'create_subprocess_exec', new=fake_exec)
-    mocker.patch.object(swtask, 'max_timeout', new=noop_async)
 
     status = await swtask.run_task(context)
     log_file = log.get_log_filename(context)
     contents = read(log_file)
     assert contents == "Automation Error: python exited with signal -11\n"
 
 
+@pytest.mark.asyncio
+async def test_run_task_timeout(context):
+    """`run_task` raises `ScriptWorkerTaskException` and kills the process
+    after exceeding `task_max_timeout`.
+    """
+    temp_dir = os.path.join(context.config['work_dir'], "timeout")
+    context.config['task_script'] = (
+        sys.executable, TIMEOUT_SCRIPT, temp_dir
+    )
+    # With shorter timeouts we hit issues with the script not managing to
+    # create all 6 files
+    context.config['task_max_timeout'] = 5
+
+    pre = arrow.utcnow().timestamp
+    with pytest.raises(ScriptWorkerTaskException):
+        await swtask.run_task(context)
+    post = arrow.utcnow().timestamp
+    # I don't love these checks, because timing issues may cause this test
+    # to be flaky. However, I don't want a non- or long- running test to pass.
+    # Did this run at all?
```

Comment:
could python freezegun help here?

Reply:
I may be misreading, but afaict, freezegun makes the datetime calls return a predictable time. How would we know if our timeout task took too long or short if both calls return the same timestamp?

Reply:
I was imagining using it for https://github.com/spulec/freezegun#manual-ticks and/or https://github.com/spulec/freezegun#moving-time-to-specify-datetime

Reply:
The time tests in this test are specifically to measure how long we're running the task for before killing: we need to know that the task runs for at least as long as max timeout, but not significantly longer (e.g., if the task runs to completion without being killed, that should be an error). Manually changing the timestamps to pass means those tests do nothing useful, and we should remove them. I know that timestamp tests are potentially flaky, but I also want to know if the kill isn't working properly.

Reply:
Fair, I was musing that it may help, not blocking on it, to be extra clear

Reply:
Ok. Thanks for the pointer; looks like that should be helpful in the future for other time-based tests.

I'm going to push scriptworker 14.0.0 changes here and then merge.